### DISCUSSION TITLE
Middleware to intercept requests and add a specific id?

### DISCUSSION THREAD

**Discussion by jmcelreavey** [prisma/prisma#7205] [Q&A]
I'm attempting to write Prisma middleware which will intercept the request and ensure that every model which is queried or updated has an organisationId in the data or where clause. 

This has turned out more difficult than expected as if someone uses a connect instead of an ID then adding the organisationID will result in an error. Also if someone connects on to other entities or creates other entities of a parent one we have no way to ensure the organisationID is inserted.

Is there any way that I can ensure that no matter what happens in the database if an update takes place on a table the organisationID is added. (currently exists on the user table and is always available in the API)

**Comment by janpio** [comment]
Do you have an example for these problems you are experiencing? Hard to image exactly from the description.

**Reply by jmcelreavey** [reply]
Apologies, here's a not so nice piece of code I've been using for this problem:

```
    prisma.$use(async (params, next) => {
        // Check incoming query type to see if we exclude it or not
        if (params.model && unscopedModels.indexOf(params.model) === -1) {
            // Check for insertions
            if (params.action === "create" || params.action === "createMany") {
                // Ensure we set the organisation, this will override if it's already set
                params.args["data"] = { ...params.args["data"], organisationId: organisationId };
            }
            // Updates
            else if (params.action === "upsert" || params.action === "update" || params.action === "delete") {
                if (params.action === "upsert") {
                    params.args["create"] = { ...params.args["create"], organisationId: organisationId };
                    params.args["update"] = { ...params.args["update"], organisationId: organisationId };
                } else if (params.action === "update") {
                    params.args["data"] = { ...params.args["data"], organisationId: organisationId };
                }

                params.args["where"] = JSON.parse(createWhereKey(params, organisationId), function (k, v) {
                    return typeof v === "object" || isNaN(v) ? v : parseInt(v, 10);
                });
            } else {
                params.args["where"] = { ...params.args["where"], organisationId: organisationId };
            }
        }

        return next(params);
    });
```

Essentially I take every request and add the organisationId to the data and where clause to ensure that any updates sets the organisationId accordingly. (every model has an organisationId). Multiple customers will be using the same database so I want to ensure that they're scoped to there organisation without the other developers having to think too much about it.

I don't think it's going to be possible though becauses issue arise when they try to connect to other models or create children as part of the request. If they use an checkedArgsInput too we wouldn't be able to just append the organisationId as that will throw an exception, we need to use the "Organisation" object.

Hope that makes some more sense.

Essentially I want all my data scoped to an organisation automatically based off the logged in users organisationId which we pass about on the token requests
